api.md

```diff
@@ -266,71 +266,157 @@ Output:
 
 Status Code: 200 or 400
 
+### Create Project
+
+POST "/project_create"
+
+This will create a project with the current user as the admin.
+
+Only requires name and description of the project.
+
+Input: A JSON body with the following **required** parameters.
+
+```typescript
+{
+    name: string;
+    description: string;
+}
+```
+
+Status Code: 201 or 400
+
+### Get Project
+
+GET "/project_get"
+
+This will get the project's name, description and creation time.
+
+Input: A JSON body with the following **required** parameters.
+
+```typescript
+{
+    projectid: string;
+}
+```
+
+Status Code: 200 or 400
+
+### Create Task
+
+POST "/task_create"
+
+This will create either a personal task or project task based on if a project ID is passed in.
+No projectid passed in, the task will be created for current user.
+If specified, will create a task for the project and assign users in user array to that task.
+
+Input: A JSON body with the following **required** parameters.
+
+```typescript
+{
+  name: string;
+  description: string;
+  users: string[];
+  projectID: string;
+}
+```
+
+### Add User to Task
+
+PATCH "/task_add_user"
+
+This will add users to the aforementioned task.
+Only call on tasks in projects.
+All users in the users array will be given this task.
+
+Input: A JSON body with the following **required** parameters.
+
+```typescript
+{
+  taskid: string;
+  users: string[];
+}
+```
+
+### Delete Task
+
+PATCH "/task_delete"
+
+Deletes all tasks that are given. Provide projectid if its a task belonging to a project.
+
+Input: A JSON body with the following **required** parameters.
+
+```typescript
+{
+  projectid: string;
+  tasks: string[];
+}
+```
+
```

Comment:
misc stuff:

-> need to write the exact outputs (otherwise idk the exact name) of the requests (similarly to the inputs)

-> also do you want to handle creationTime on server side or client side? (i assume its server side because you don't take it in?)

Reply:
Yah creation time is handled server side